irods/manager/user_manager.py

```diff
@@ -187,25 +202,42 @@ def get(self, name, user_zone=""):
             raise UserGroupDoesNotExist()
         return iRODSUserGroup(self, result)
 
-    def create(self, name, user_type='rodsgroup', user_zone="", auth_str="", group_admin=False):
-        if group_admin:
-            message_body = UserAdminRequest(
+    def _api_info(self, group_admin_flag):
+        """
+        If group_admin_flag is:   then use UserAdminRequest API:
+        --------------------      ------------------------------
+        True                      always
+        False                     never
+        None                      when user type is groupadmin
```

Comment:
I'm not sure if this is a good thing or not. My gut tells me this could lead to hard-to-reason about bugs in a complex application, but I'd like to know what the motivation is before jumping to conclusions.

Please explain why you feel this is good?
Is it common for the PRC to invoke different APIs based on info in the session?

Reply:
To me, the APIs are a means to an end, I guess. It's the difference between two cases for how to write a utility script that (for example) manipulates users and groups.  Suppose you want to write such an application, but the `group_admin` flag is always there as a consideration.  Example app:
```
extra_opts = { 'group_admin': False }
if i_am_a_group_admin:
    extra_opts['group_admin'] = True
session.user_groups.create ( groupname , **extra_opts )
for userX in user_names:
    session.user_groups.addmember (groupname, userX, **extra_opts)
```
That, vs. simply this, if the API were optionally to default to the appropriate one given the user type:
```
session.user_groups.create (groupname)
for userX in user_names:
    session.user_groups.addmember (groupname, userX)
```
I don't have anything invested in this, other than the library being more approachable to new or possibly users... if it rankles, I can easily leave it be.  I was just thinking of this as an opportunity (since we may soon be renaming the interface from `user_groups` to just `groups`) to have a slightly less strict default.  If we want to force the API point, we can always pass `group_admin` as `True` or `False`. @MartinFlores751  ? @trel? @alanking ?

Reply:
@korydraughn - But, other than this, no... I can't think of another place where the PRC chooses API based on the situation, unless the multi-1247 parallel transfer code qualifies.  It will - based on file or object size choose either the single threaded open/read-or-write/close codepath or the multithreaded codepath that uses replica access tokens, close replica, and all that.

Reply:
Consensus reached; we keep this as-is for PRC and probably consider how better to separate high-level / low-level operations for the library re-design, whenever that happens.